### ISSUE TITLE
Deprecated tag support for JavaScript and TypeScript in overrides in children

### ISSUE THREAD

**Issue by thernstig** [microsoft/TypeScript#42823]
Issue Type: <b>Feature Request</b>

https://code.visualstudio.com/updates/v1_49#_deprecated-tag-support-for-javascript-and-typescript added the great `@deprecated` tag support which works great.

However, when extending a parent class that has a `@deprecated` method, that method is not shown with strikethrough in the child. It would be great if this was the case.

VS Code version: Code 1.53.2 (622cb03f7e070a9670c94bae1a45d78d7181fbd4, 2021-02-11T11:48:04.245Z)
OS version: Windows_NT x64 10.0.18363
Remote OS version: Linux x64 5.4.0-65-generic

**Comment by thernstig** [comment]
Repro:

```js
// father.js

class Father {
  constructor() {
    console.log('constructor');
  }

  /**
   * @returns {string} stuff
   */
  newFunction() {
    return 'new';
  }

  /**
   * @deprecated
   *
   * @returns {string} stuff
   */
  oldFunction() {
    return 'old';
  }
}

module.exports = { Father };
```

```js
// child.js

const { Father } = require('./father');

class Child extends Father {
  /**
   * @returns {string} stuff
   */
  newFunction() {
    return 'new child';
  }

  /**
   * @returns {string} stuff
   */
  oldFunction() {
    return 'old child';
  }
}

module.exports = { Child };
```

Notice how `oldFunction` does not have strikethrough. Also, hovering that function in both the father and the child does not show it as strikethrough in the popup.

However, during autocompletion when writing the function name it properly shows as strikethrough:
![child_intellisense_autocompletion](https://user-images.githubusercontent.com/30827238/108181074-c2b0ab80-7107-11eb-8efb-e4976f01b338.png)

**Comment by RyanCavanaugh** [comment]
This seems subjective? The base class method being deprecated doesn't necessarily imply the derived class method is too.

**Comment by thernstig** [comment]
True. I wonder what the best option is here. As can be seen in the IntelliSense it currently does use strikethrough.
